### ISSUE TITLE
write_fmt return number of bytes written

### ISSUE THREAD

**Issue by MabezDev** [rust-lang/rust#63614]
The current [`write_fmt`](https://doc.rust-lang.org/std/fmt/trait.Write.html#method.write_fmt) implementation has a return type of `Result<()>` is there any technical reason it couldn't return the number of bytes written? Similar to sprintf in C.

An example use case in Smoltcp:
```rust
socket.send(|mut buffer| {
    let bytes_written = write!(buffer, "Http Error {:?}", e).unwrap();
    (bytes_written, ())
}).unwrap();
```
The return type of the closure requires indicates the number of bytes used such that smoltcp can flush those bytes to the wire.

**Comment by jonas-schievink** [comment]
I don't think there's a technical reason for it, but changing it would be a breaking change, so it's not going to happen.

**Comment by MabezDev** [comment]
Fair enough. Is there anyway to use the formatter and get the number of bytes it will create? I really want to avoid hand rolling formatting if possible.

**Comment by jonas-schievink** [comment]
The only way I can think of is wrapping the formatter and keeping your own count

**Comment by MabezDev** [comment]
It's a bit awkward, but it works. Thanks!
